packages/next/server/web/sandbox/sandbox.ts

```diff
@@ -221,3 +232,16 @@ function getFetchURL(input: RequestInfo, headers: NodeHeaders = {}): string {
 function isRequestLike(obj: unknown): obj is Request {
   return Boolean(obj && typeof obj === 'object' && 'url' in obj)
 }
+
+function __next_eval__(fn: Function) {
+  const key = fn.toString()
+  if (!cache?.warnedEvals.has(key)) {
+    console.warn(
+      new Error(
```

Comment:
Can we add the path to the `_middleware` that is triggering this to be shown here? Seems this is what is currently shown which can make it a little tricky to narrow in on which `_middleware` is causing it:

```sh
/Users/jj/dev/vercel/next.js/packages/next/server/web/sandbox/sandbox.ts:240
      new Error(
      ^

Error: Dynamic Code Evaluation (e. g. 'eval', 'new Function') not allowed in Middleware
    at null.__next_eval__ (/Users/jj/dev/vercel/next.js/packages/next/server/web/sandbox/sandbox.ts:240:7)
    at usingEval (webpack-internal:///./lib/utils.js:9:16)
    at Object.middleware [as handler] (webpack-internal:///./pages/_middleware.js:9:103)
    at adapter (webpack-internal:///../../../../packages/next/dist/server/web/adapter.js:30:35)
    at Module.__WEBPACK_DEFAULT_EXPORT__ (webpack-internal:///../../../../packages/next/dist/build/webpack/loaders/next-middleware-loader.js?page=%2F_middleware&absolutePagePath=%2FUsers%2Fjj%2Fdev%2Fvercel%2Fnext.js%2Ftest%2Fintegration%2Fmiddleware%2Fwith-eval%2Fpages%2F_middleware.js!:17:87)
    at Object.run (/Users/jj/dev/vercel/next.js/packages/next/server/web/sandbox/sandbox.ts:155:28)
    at DevServer.runMiddleware (/Users/jj/dev/vercel/next.js/packages/next/server/next-server.ts:105:36)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at DevServer.runMiddleware (/Users/jj/dev/vercel/next.js/packages/next/server/dev/next-dev-server.ts:517:22)
    at Object.fn (/Users/jj/dev/vercel/next.js/packages/next/server/next-server.ts:1139:22)
```

Reply:
Done but all credit goes to @sokra